src/rpc/blockchain.cpp

```diff
@@ -810,8 +810,10 @@ static RPCHelpMan pruneblockchain()
         height = chainHeight - MIN_BLOCKS_TO_KEEP;
     }
 
-    PruneBlockFilesManual(active_chainstate, height);
     const CBlockIndex& block{*CHECK_NONFATAL(active_chain.Tip())};
+    if (!(block.nStatus & BLOCK_HAVE_DATA)) throw JSONRPCError(RPC_MISC_ERROR, "Nothing to prune. No block data available for the chain tip.");
```

Comment:
In commit "make GetFirstStoredBlock assert that 'start_block' always has data" (aad83e489a9f515a1da36acd3e99f35b1a0da53c)

The `pruneblockchain` RPC doesn't consider it an error to prune a block that's already been pruned. It just makes a best effort to prune what it can, and then it returns the last height of block without transaction data.

So I think it would be more consistent to not make this an error, and just do:

```c++
PruneBlockFilesManual(active_chainstate, height);
return block.nStatus & BLOCK_HAVE_DATA ? GetFirstStoredBlock(block)->nHeight -1 : block->nHeight;
```

This would be a smaller change in behavior. It would make the RPC method consistently return the height of the highest block without data, instead of being off-by-one in this case.

Reply:
done 👍🏼.